src/aspire/utils/BlockDiagonal.pyx

```diff
@@ -0,0 +1,561 @@
+import numpy as np
+
+from numpy.linalg import norm
+from numpy.linalg import solve
+from scipy.linalg import block_diag
+from scipy.special import jv
+
+from aspire.utils.cell import Cell2D
+from aspire.basis.ffb_2d import FFBBasis2D
+from aspire.basis.basis_utils import lgwt
+
+from aspire.utils.blk_diag_func import blk_diag_apply
+from aspire.utils.blk_diag_func import blk_diag_solve
+
+SCALAR_TYPES=(int, float, complex)
+
+class BlockDiagonal:
+
```

Comment:
Add initial docs string even a simple summary.

Reply:
Yes, I forgot to carry that over, thanks!